arangod/Aql/OptimizerRules.cpp

```diff
@@ -7712,9 +7714,27 @@ void arangodb::aql::moveFiltersIntoEnumerateRule(
         TRI_ASSERT(!expr->willUseV8());
         found.clear();
         Ast::getReferencedVariables(expr->node(), found);
-        if (found.size() == 1 && found.find(outVariable) != found.end()) {
-          calculations.emplace(calculationNode->outVariable(), calculationNode);
+        if (found.find(outVariable) != found.end()) {
+          // check if the introduced variable refers to another temporary
+          // variable that is not valid yet in the EnumerateCollection/Index
```

Comment:
The "move-filters-into-enumerate" rule never took views into account. It can move filter conditions into IndexNodes and EnumerateCollectionNodes.
I am happy to extend this to views as well if it is possible. I guess this could be done by fusing a view's existing search condition with other filters that we find along the way?